src/objective.jl

```diff
@@ -160,3 +160,36 @@ function objective_function(model::Model,
                    MOI.ObjectiveFunction{MOIFunType}())::MOIFunType
     return jump_function(model, func)
 end
+
+"""
+    set_objective_coefficient(model::Model, variable::VariableRef, coefficient::Real)
+
+Set the linear objective coefficient associated with `Variable` to `coefficient`.
+
+Note: this function will throw an error if a nonlinear objective is set.
+"""
+function set_objective_coefficient(model::Model, variable::VariableRef, coeff::Real)
+    if model.nlp_data !== nothing && _nlp_objective_function(model) !== nothing
+        error("A nonlinear objective is already set in the model")
+    end
+
+    obj_fct_type = moi_function_type(objective_function_type(model))
+    if obj_fct_type == MOI.SingleVariable
+        # Promote the objective function to be an affine expression.
+        current_obj = objective_function(model)
+        if index(current_obj) == index(variable)
+            set_objective_function(model, coeff * variable)
+        else
+            set_objective_function(model, add_to_expression!(coeff * variable, current_obj))
```

Comment:
This should be `add_to_expression!(current_obj, coef, variable)`. The expression being added to is the first argument.

Reply:
If the `else` reduces to `add_to_expression!(current_obj, coeff, variable)`, then I get this "nice" method error: 

```
  Got exception outside of a @test
  MethodError: no method matching add_to_expression!(::VariableRef, ::Float64, ::VariableRef)
  Closest candidates are:
    add_to_expression!(!Matched::GenericAffExpr{C,V}, ::Real, ::V) where {C, V} at C:\Users\…\.julia\dev\JuMP\src\aff_expr.jl:236
    add_to_expression!(!Matched::GenericQuadExpr{C,V}, ::Real, ::V) where {C, V} at C:\Users\…\.julia\dev\JuMP\src\quad_expr.jl:174
    add_to_expression!(!Matched::GenericQuadExpr{C,V}, ::C, ::V, !Matched::V) where {C, V} at C:\Users\…\.julia\dev\JuMP\src\quad_expr.jl:278
    ...
  Stacktrace:
   [1] set_objective_coefficient(::Model, ::VariableRef, ::Float64) at C:\Users\…\.julia\dev\JuMP\src\objective.jl:183
```

Reply:
Oh, good point. LGTM as is.